include/libcyphal/config.hpp

```diff
@@ -0,0 +1,190 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_CONFIG_HPP_INCLUDED
+#define LIBCYPHAL_CONFIG_HPP_INCLUDED
+
+#include <cstddef>
+
+namespace libcyphal
+{
+
+// All below NOSONAR cpp:S799 "Rename this identifier to be shorter or equal to 31 characters."
+// are intentional and are used to keep the names consistent with the rest of the codebase.
+// F.e. `IExecutor_Callback_FunctionMaxSize` is consistent with `IExecutor::Callback::FunctionMaxSize`.
+
+/// Defines various configuration parameters of libcyphal.
+///
+/// All methods are `static constexpr` - they are evaluated at compile time.
+///
+/// Nolint b/c this is the main purpose of this file - define various constants.
+/// NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+///
+struct Config
+{
+    /// Defines max footprint of a callback function in use by the executor.
+    ///
+    static constexpr std::size_t IExecutor_Callback_FunctionMaxSize()  // NOSONAR cpp:S799
```

Comment:
Why are these functions instead of simple `static constexpr` constants? iirc even in C++14 these can be defined without out-of-line definitions.

Reply:
ODR problem in c++14 if you pass such constant to somewhere which needs address of it (like `std::max` f.e.).